### ISSUE TITLE
Extra properties does not trigger error when spreading object

### ISSUE THREAD

**Issue by hansott** [microsoft/TypeScript#43499]
# Bug Report

```ts
function inferType(value: string) {
    const asBoolean: boolean | null = null;
    const asDate: Date | null = null;
    const asDouble: number | null = null;
    const asLong: number | null = null;
    const asURL: URL | null = null;

    return {
        value_as_string: value,
        value_as_boolean: asBoolean,
        value_as_date: asDate,
        value_as_double: asDouble,
        value_as_long: asLong,
        value_as_url: asURL,
  };
}

type EventMetaData = {
  key: string;
  value_as_string: string;
  value_as_boolean: boolean | null;
  value_as_date: string | null;
  value_as_long: number | null;
  value_as_double: number | null;
};

function convert(key: string, value: string): EventMetaData {
    const types = inferType(value);

    return {
        key: key,
        ...types // Expect error here because `value_as_url` is not allowed
    };
}
```

Would expect the following error (or something like that):

```
Type '{ value_as_url: URL; value_as_string: string; value_as_boolean: null; value_as_date: null; value_as_double: null; value_as_long: null; key: string; }' is not assignable to type 'EventMetaData'.
  Object literal may only specify known properties, and 'value_as_url' does not exist in type 'EventMetaData'.(2322)
```

### 🔎 Search Terms

object spread extra properties expected error

### 🕗 Version & Regression Information

Using TypeScript 4.1.5

### ⏯ Playground Link

[No error playground](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210402#code/GYVwdgxgLglg9mABDMwCmAnAKgTwA5oAUAbgIYA2IaAXIgM5QYoDmAlIgN4BQiviECBolJ0AQnDjk0pMLQBGEqTMQAfRGBDlyiALzrN5ANw8+AsEJEARUlBqJrt1fq27nRk7zMW6luCDlStBoAtnKYThouepHufPyCUMJ0ADIIzEEgoeFqMa4xxnFeiSIAqgBKybTlyREGeQbGHogYaFAgGEjccXFklGgA+iL9DExg6Yi9VAA0TT0UVIN0-QqS0rJJ4qsyM918kwNDACY2dlYnO7sT8wdLh34Bpz73Uhe7+4v95Gm0Iqljr913kN2uQfnRqhcAL7GSFcLhQfBoRAAUWIaDAUAAsq1SA5SK4uogANZoHC0EYsApXPofCljcmMSkmIFLFZKdZsta1LRUln9Y62BmjZjc2LUhZDL70-RZDCi3nXD53fyBGVhOU5BpcaFw0CQWAIeJgNEYKCEElk+iM-7iux0ti0VHorE4vGcJpFRAIgh0VwodDYREka6sRpxFptDruy7E0m0C0AuIAOhT3rQdCaOshQA)

[Expected error](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210402#code/GYVwdgxgLglg9mABDMwCmAnAKgTwA5oAUAbgIYA2IaAXIgM5QYoDmAlIgN4BQiviECBolJ0AQnDjk0pMLQBGEqTMQAfRGBDlyiALzrN5ANw8+AsEJEARUlBqJrt1fq27nRk7zMW6luCDlStBoAtnKYThouepHufPyCUMJ0ADIIzEEgoeFqMa4xxnFeiSIAqgBKybTlyREGeQbGHogYaFAgGEjccXFklGgA+iL9DExg6Yi9VAA0TT0UVIN0-QqS0rJJ4qsyM918kwNDACY2dlYnO7sT8wdLh34Bpz73Uhe7+4v95Gm0Iqljr913kN2uQfnRqhcAL7GSFcLhQfBoRAAUWIaDAUAAsq1SA5SK4uogANZoHC0EYsApXPofCljcmMSkmIFLFZKdZsta1LRUln9Y62BmjZjc2LUhZDL70-RZDCi3nXD53fyBGVhOU5BpcaFw0CQWAIeJgNEYKCEElk+iM-7iux0ti0VHorE4vGcJpFRAIgh0VwodDYREka6sRpxFptDruy7E0m0C0AuIAOhT3rQdETe0VwIwoPUaAA7ohqoQAESl1gAnWQoA)

**Comment by MartinJohns** [comment]
This is intentional: #19775

**Comment by hansott** [comment]
@MartinJohns I see, thanks! 

Subscribing to https://github.com/microsoft/TypeScript/issues/12936 ;)
